### ISSUE TITLE
Docker nightly raspbian buster package doesn't install on armv6 arch (Pi Zero, Pi)

### ISSUE THREAD

**Issue by ndanyluk** [moby/moby#39542]
**Description**

I've observed that the last working Docker package for Raspbian Stretch/Buster for armv6 devices is 18.06.3.  I was hoping with the move to Buster and 19.03.0 that this would be resolved, but after trying to install on my Docker swarm of Pi Zeros last night I found that the issue persists.

Given the Buster sources list does not have version 18.06.3 to fall back to, this renders Docker unusable for Pi Zero, Pi Zero W, and original Pi devices (Pi 3 and above are not affected)

**Steps to reproduce the issue:**
1. Flash a Pi Zero or Pi/Pi 2 with Raspbian Buster
2. Attempt to install Docker with the convenience script or by manually editing the sources.list
3. Attempt to install Docker with apt

**Describe the results you received:**
Docker is unable to run on my Pi Zero at all (service instantly fails)

**Describe the results you expected:**
Docker service running and daemon accessible over CLI

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Version:           0.0.0-20190714170256-22f3318
 API version:       1.40
 Go version:        go1.12.6
 Git commit:        22f3318
 Built:             Mon Jul 15 07:42:25 2019
 OS/Arch:           linux/arm
 Experimental:      false
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

**Output of `docker info`:**

```
Client:
 Debug Mode: false

Server:
ERROR: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
errors pretty printing info
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Error only occurs on ARMv6 architecture (still classified as armhf by Raspbian)

**Output of `service docker status`**

```
● docker.service - Docker Application Container Engine
   Loaded: loaded (/lib/systemd/system/docker.service; enabled; vendor preset: enabled)
   Active: inactive (dead) since Tue 2019-07-16 19:16:53 EDT; 2s ago
     Docs: https://docs.docker.com
  Process: 25565 ExecStart=/usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock (code=killed, signal=TERM)
 Main PID: 25565 (code=killed, signal=TERM)

Jul 16 19:16:50 pi-five systemd[1]: Starting Docker Application Container Engine...
Jul 16 19:16:53 pi-five systemd[1]: docker.service: Main process exited, code=killed, status=15/TERM
Jul 16 19:16:53 pi-five systemd[1]: docker.service: Succeeded.
Jul 16 19:16:53 pi-five systemd[1]: Stopped Docker Application Container Engine.
```

**Comment by ndanyluk** [comment]
Here is an update with the output of `apt upgrade docker-ce docker-ce-cli` since I didn't think to keep the initial logs.  The errors look the same though

```
Reading changelogs... Done
(Reading database ... 92245 files and directories currently installed.)
Preparing to unpack .../docker-ce-cli_5%3a0.0.0-20190717011415-0cd65b8-0~raspbian-buster_armhf.deb ...
Unpacking docker-ce-cli (5:0.0.0-20190717011415-0cd65b8-0~raspbian-buster) over (5:0.0.0-20190714170256-22f3318-0~raspbian-buster) ...
Preparing to unpack .../docker-ce_5%3a0.0.0-20190717011415-0cd65b8-0~raspbian-buster_armhf.deb ...
Unpacking docker-ce (5:0.0.0-20190717011415-0cd65b8-0~raspbian-buster) over (5:0.0.0-20190714170256-22f3318-0~raspbian-buster) ...
Setting up docker-ce-cli (5:0.0.0-20190717011415-0cd65b8-0~raspbian-buster) ...
Setting up aufs-dkms (4.19+20190211-1) ...
Removing old aufs-4.19+20190211 DKMS files...

------------------------------
Deleting module version: 4.19+20190211
completely from the DKMS tree.
------------------------------
Done.
Loading new aufs-4.19+20190211 DKMS files...
It is likely that 4.19.50+ belongs to a chroot's host
Building for 4.19.57+, 4.19.57-v7+ and 4.19.57-v7l+
Building initial module for 4.19.57+
Error! Bad return status for module build on kernel: 4.19.57+ (armv6l)
Consult /var/lib/dkms/aufs/4.19+20190211/build/make.log for more information.
dpkg: error processing package aufs-dkms (--configure):
 installed aufs-dkms package post-installation script subprocess returned error exit status 10
Setting up docker-ce (5:0.0.0-20190717011415-0cd65b8-0~raspbian-buster) ...
Job for docker.service canceled.
invoke-rc.d: initscript docker, action "start" failed.
● docker.service - Docker Application Container Engine
   Loaded: loaded (/lib/systemd/system/docker.service; enabled; vendor preset: enabled)
   Active: inactive (dead) since Wed 2019-07-17 11:47:09 EDT; 678ms ago
     Docs: https://docs.docker.com
  Process: 5604 ExecStart=/usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock (code=killed, signal=TERM)
 Main PID: 5604 (code=killed, signal=TERM)

Jul 17 11:47:07 pi-five systemd[1]: Starting Docker Application Container Engine...
Jul 17 11:47:09 pi-five systemd[1]: docker.service: Main process exited, code=killed, status=15/TERM
Jul 17 11:47:09 pi-five systemd[1]: docker.service: Succeeded.
Jul 17 11:47:09 pi-five systemd[1]: Stopped Docker Application Container Engine.
dpkg: error processing package docker-ce (--configure):
 installed docker-ce package post-installation script subprocess returned error exit status 1
Processing triggers for man-db (2.8.5-2) ...
Processing triggers for systemd (241-5+rpi1) ...
Errors were encountered while processing:
 aufs-dkms
 docker-ce
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Comment by Jaimenms** [comment]
I have the same problem. It seems to be related to containerd.

**Comment by jasperdunn** [comment]
I'm also unable to install docker using the convenience scripts or apt onto raspbian:buster on a RaspberryPi 4

**Comment by thaJeztah** [comment]
I think this should be resolved with the latest 1.2.10 containerd.io package
